php/libraries/Database.class.inc

```diff
@@ -67,34 +67,37 @@ class Database
 
     /**
      * Singleton design pattern implementation - creates the object
-     * and also connects to the database if necessary
+     * and also connects to the database if necessary.
+     * NOTE There is no return type declaration for this function as it returns
+     * the type `Database` which is not supported by PHPs built in return types.
      *
      * @param string  $database     the database to select
      * @param string  $username     the username with which to log into
-     *                              the database server
+     *                                      the database server
      * @param string  $password     the password that matches the username
      * @param string  $host         the name of the database server
      * @param boolean $trackChanges boolean determining if changes should be
      *                              logged to the history table
      *
-     * @return Database
+     * @return object Database
      * @throws DatabaseException
      * @access public
      */
     static function &singleton(
-        $database=null,
-        $username=null,
-        $password=null,
-        $host=null,
-        $trackChanges=true
+        string $database     = '',
```

Comment:
Think so. Using `null` was actually less safe because this variable is meant to be a string. I changed the if statement just below it to check for `empty` instead of `null` and I believe that's the only place we need to worry about it.